### PR TITLE
Convert pluginsv2 to pure containerd client

### PR DESCRIPTION
**Pull request by crosbymichael**
This moves the oci and caps package under a new top level runtime package.  This will be the home for runtime related containerd Opts and spec values for the upcoming work. 

This also converts the plugin code to use the containerd package directly instead of libcontainerd.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@crosbymichael CI is failing on some linting issues;

```
18:18:25 plugin/executor/containerd/containerd_test.go:45:37:warning: undeclared name: Client (gosimple)
18:18:25 plugin/executor/containerd/containerd.go:87:47:warning: stderr can be io.Writer (interfacer)
18:18:25 plugin/executor/containerd/containerd.go:87:39:warning: stdout can be io.Writer (interfacer)
18:18:25 plugin/executor/containerd/containerd.go:43:63:warning: stderr can be io.Writer (interfacer)
18:18:25 plugin/executor/containerd/containerd.go:43:55:warning: stdout can be io.Writer (interfacer)
```

**Comment by crosbymichael** [comment]
@thaJeztah i know;)

I didnt' know warnings were failures....

File: plugin/executor/containerd/containerd.go [hunk 1]
```diff
@@ -25,153 +23,141 @@ type ExitHandler interface {
...
 	if err != nil {
-		deleteTaskAndContainer(ctx, e.client, id)
+		container.Delete(ctx)
+		return err
 	}
-	return err
+	wait, err := task.Wait(ctx)
+	if err != nil {
+		task.Delete(ctx)
+		container.Delete(ctx)
```
Comment (thaJeztah) [inline plugin/executor/containerd/containerd.go]: Errors returned by `task.Delete()` and `container.Delete()` don't have to be handled? (Here, and other cases below).

**Review by thaJeztah — APPROVED** [review]
LGTM, but left a question

**Comment by tiborvass** [comment]
Are the removed tests covered by some other ones?

**Comment by crosbymichael** [comment]
@tiborvass i'm still having some issues with the docker-py tests with these changes.

As for the current tests in that dir, I looked at them and didn't find any value with the changes.  It was just testing that a mock's methods were being called and nothing more.

**Review by cpuguy83 — COMMENTED** [review]
Yay!

One thing that libcontainerd did was convert errors from containerd errors into errors from the errdefs package in moby.

**Comment by crosbymichael** [comment]
@cpuguy83 maybe that is the issues i was seeing, something above expects a docker errdef

**Comment by cpuguy83** [comment]
p.s., the only reason this executor package exists is because libcontainerd has all kinds of platform dependent code which made it difficult to work on the plugin package and made some really weird contortions for creating/installing plugins for integrations tests.

**Comment by crosbymichael** [comment]
@cpuguy83 do you think we should just remove that package all together now?

**Comment by cpuguy83** [comment]
It could certainly be removed as separate work.

**Pull request closed without merging**